File: lib/src/main/java/com/yalantis/contextmenu/lib/AnimatorUtils.java
package com.yalantis.contextmenu.lib;

import android.view.View;

import com.nineoldandroids.animation.AnimatorSet;
import com.nineoldandroids.animation.ObjectAnimator;

public class AnimatorUtils {

    public static ObjectAnimator rotationCloseToRight(View v) {
        return ObjectAnimator.ofFloat(v, "rotationY", 0, -90);
    }

    public static ObjectAnimator rotationOpenFromRight(View v) {
        return ObjectAnimator.ofFloat(v, "rotationY", -90, 0);
    }

    public static ObjectAnimator rotationCloseVertical(View v) {
        return ObjectAnimator.ofFloat(v, "rotationX", 0, -90);
    }

    public static ObjectAnimator rotationOpenVertical(View v) {
        return ObjectAnimator.ofFloat(v, "rotationX", -90, 0);
    }

    public static ObjectAnimator alfaDisappear(View v) {
        return ObjectAnimator.ofFloat(v, "alpha", 1, 0);
    }

    public static ObjectAnimator alfaAppear(View v) {
        return ObjectAnimator.ofFloat(v, "alpha", 0, 1);
    }

    public static ObjectAnimator translationRight(View v, float x) {
        return ObjectAnimator.ofFloat(v, "translationX", 0, x);
    }
    public static ObjectAnimator translationLeft(View v, float x) {
        return ObjectAnimator.ofFloat(v, "translationX", x, 0);
    }

    public static AnimatorSet fadeOutSet(View v, float x){
        AnimatorSet fadeOutSet = new AnimatorSet();
        fadeOutSet.playTogether(alfaDisappear(v), translationRight(v,x));
        return fadeOutSet;
    }

}

File: lib/src/main/java/com/yalantis/contextmenu/lib/AnimatorUtils.kt
package com.yalantis.contextmenu.lib

import android.view.View
import com.nineoldandroids.animation.AnimatorSet
import com.nineoldandroids.animation.ObjectAnimator

object AnimatorUtils {

    private const val ROTATION_Y_PROPERTY = "rotationY"
    private const val ROTATION_X_PROPERTY = "rotationX"
    private const val ALPHA_PROPERTY = "alpha"
    private const val TRANSLATION_X_PROPERTY = "translationX"

    @JvmStatic
    fun rotationCloseToRight(view: View): ObjectAnimator =
            ObjectAnimator.ofFloat(view, ROTATION_Y_PROPERTY, 0f, -90f)

    @JvmStatic
    fun rotationOpenFromRight(view: View): ObjectAnimator =
            ObjectAnimator.ofFloat(view, ROTATION_Y_PROPERTY, -90f, 0f)

    @JvmStatic
    fun rotationCloseVertical(view: View): ObjectAnimator =
            ObjectAnimator.ofFloat(view, ROTATION_X_PROPERTY, 0f, -90f)

    @JvmStatic
    fun rotationOpenVertical(view: View): ObjectAnimator =
            ObjectAnimator.ofFloat(view, ROTATION_X_PROPERTY, -90f, 0f)

    @JvmStatic
    fun alphaDisappear(view: View): ObjectAnimator =
            ObjectAnimator.ofFloat(view, ALPHA_PROPERTY, 1f, 0f)

    @JvmStatic
    fun alphaAppear(view: View): ObjectAnimator =
            ObjectAnimator.ofFloat(view, ALPHA_PROPERTY, 0f, 1f)

    @JvmStatic
    fun translationRight(view: View, x: Float): ObjectAnimator =
            ObjectAnimator.ofFloat(view, TRANSLATION_X_PROPERTY, 0f, x)

    @JvmStatic
    fun translationLeft(view: View, x: Float): ObjectAnimator =
            ObjectAnimator.ofFloat(view, TRANSLATION_X_PROPERTY, x, 0f)

    @JvmStatic
    fun fadeOutSet(view: View, x: Float): AnimatorSet = AnimatorSet().apply {
        playTogether(alphaDisappear(view), translationRight(view, x))
    }
}